activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -130,7 +130,7 @@ def self.add_default_callbacks(model, reflection)
     end
 
     def self.add_destroy_callbacks(model, reflection)
-      model.after_destroy lambda { |o| o.association(reflection.name).handle_dependency }
+      model.before_destroy lambda { |o| o.association(reflection.name).handle_dependency }
```

Comment:
@fegorodscy is this why the circular dependency test was failing?

Reply:
That's to prevent `UncaughtThrowError: uncaught throw :abort`.
All other associations handle dependency before_destroy, only belongs_to had this being handled after.

Reply:
Actually this change break things for DB's other than sqlite3. I'm investigating this further

Reply:
So, I reverted this to `after_destroy` so it won't impact other stuff (and other tests don't break).

Then, instead of `throw :abort` like the other associations, this would have to `raise ActiveRecord::Rollback`